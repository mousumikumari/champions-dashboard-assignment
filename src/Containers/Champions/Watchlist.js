import React, { useEffect, useState } from "react";
import { useHistory, Link } from "react-router-dom";
import * as ReactBootStrap from "react-bootstrap";
import withClass from '../../Hoc/withClass';


const Watchlist = (props) => {
/**
 * This Function is responsible to show the Watchlist
 *
 * @param {object} props Watchlist details
 */

  const watchlist = JSON.parse(localStorage.getItem("watchlist"));
  const [modWatchlist, setWatchlist] = useState(watchlist);
  const history = useHistory();
  const Button = () => {
  return <button
  type="button"
  className="btn btn-primary btn-lg"
>
  Remove
</button>
}
  const ColoredButton = withClass(Button);
  const {
   openChampionDetails
  } = props;
  const openHome = () => {
    history.push("/");
  };
  useEffect(() => {
    localStorage.setItem("watchlist", JSON.stringify(modWatchlist));
  });

  const removeChampion = (event) => {
    const watch = modWatchlist.filter((e) => e.id !== event);
    setWatchlist(watch, () => { localStorage.setItem("watchlist", JSON.stringify(modWatchlist)); });
  };
  return(
      <div>
    <div className="container">
    <div className="row">
      <div className="col left"> <h2>My Watchlist</h2></div>
      <div className="col right">
        <button
          type="button"
          className="styledButton"
          onClick={openHome}
        >
          Back to Home
        </button>
      </div>
    </div>
  </div>
  <div className="container">
    <ReactBootStrap.Table responsive="md" bordered hover>
      <thead>
        <tr className="center">
          <th>Champion Profile</th>
          <th>Champion ID</th>
          <th>Name</th>
          <th>Armor</th>
          <th>armorperlevel</th>
          <th>hp</th>
          <th>Watchlist Action</th>
        </tr>
      </thead>
      <tbody>
        {modWatchlist !=null && modWatchlist.map((champion) => (
          <tr key={champion.id} className="center">
            <td>
              <img src={champion.image_url} alt={champion.image_url} />
            </td>
            <td>{champion.id}</td>
            <td>
              <Link
                to={{
                  pathname: "/ChampionInfo",
                  state: { champion },
                }}
              >
                {champion.name}
              </Link>
            </td>

            <td>{champion.armor}</td>
            <td>{champion.armorperlevel}</td>
            <td>{champion.hp}</td>
            <td>
              <div>
                <button
                  type="button"
                  className="btn btn-primary btn-lg"
                  onClick={() => removeChampion(champion.id)}
                >
                  Remove
                </button>

              </div>
            </td>
          </tr>
        ))}
      </tbody>
    </ReactBootStrap.Table>
  </div>

  </div>

  );
};

export default Watchlist;