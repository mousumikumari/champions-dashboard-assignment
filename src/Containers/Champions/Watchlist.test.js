/* eslint-disable */
import React from "react";
import renderer from "react-test-renderer";
import { createMemoryHistory } from "history";
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure } from "enzyme";
import Watchlist from "./Watchlist";

const location = {
  hash: "",
  key: "hhkdz",
  pathname: "/SubmitTest",
  search: "",
  state: {
    articles: [],
    searches: [],
    watchlist: [],
    page: 1,
    pageSize: 10,
    articleLength: 50,
    visibleArticles: [],
    searchedText: "",
    champion: [],
    sortBy: "Asc",
  },
};

const state = {
  watchlist: []
}

test("First Snapshot test", () => {
  localStorage.setItem("state", JSON.stringify(state));
  const component = renderer.create(
    <Watchlist location={ location }/>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

configure({ adapter: new Adapter() });

const history = createMemoryHistory();
history.push("/");
describe("Test Button component", () => {
  it("Test click event", () => {
    const historyMock = { push: jest.fn() };
    localStorage.setItem("state", JSON.stringify(state));
    const shallowCopy = shallow(<Watchlist history={historyMock} location={location} />);
    shallowCopy.find("button").simulate("onClick");
    expect(history.location.pathname).toEqual("/");
    localStorage.removeItem("state");
  });
});