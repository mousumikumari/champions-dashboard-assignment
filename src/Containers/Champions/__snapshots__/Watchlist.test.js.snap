// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`First Snapshot test 1`] = `
<div>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col left"
      >
         
        <h2>
          My Watchlist
        </h2>
      </div>
      <div
        className="col right"
      >
        <button
          className="styledButton"
          onClick={[Function]}
          type="button"
        >
          Back to Home
        </button>
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="table-responsive-md"
    >
      <table
        className="table table-bordered table-hover"
      >
        <thead>
          <tr
            className="center"
          >
            <th>
              Champion Profile
            </th>
            <th>
              Champion ID
            </th>
            <th>
              Name
            </th>
            <th>
              Armor
            </th>
            <th>
              armorperlevel
            </th>
            <th>
              hp
            </th>
            <th>
              Watchlist Action
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>
</div>
`;
